sdk/ai/azure-ai-inference/azure/ai/inference/_patch.py

```diff
@@ -145,10 +146,142 @@ class ChatCompletionsClient(ChatCompletionsClientGenerated):
     :paramtype api_version: str
     """
 
-    def __init__(self, endpoint: str, credential: Union[AzureKeyCredential, "TokenCredential"], **kwargs: Any) -> None:
-        self._model_info: Optional[_models.ModelInfo] = None
+    def __init__(
```

Comment:
Do we plan to document the chat completion options at the client constructor?

Reply:
I would very much like to avoid that. That's why we have .with_default(). I believe it's only going to cause confusion if we provide multiple ways of doing the same thing, and exposing long lists or arguments on both constructor, with_defaults AND complete methods.